### PR TITLE
Echo and Replacer

### LINKED ISSUE #11: Handle environment variables `$TEXT`
`$` followed by a sequence of characters should expand to their values.

### LINKED ISSUE #12: Handle `$?`
Should expand to the exit status of the most recently executed foreground pipeline.

### LINKED ISSUE #13: Implement `echo` built-in
Implement `echo` with no options and with option `-n`.
### Description

Echo the **STRING**(s) to the standard output.

**`-n`** do not output the trailing newline.

### PR DESCRIPTION
**Pull request by ddiniz-m**
The branch is called heredoc, but no heredoc was worked on.

Instead I handled environment variables and echo.

The way I did it is substitute the strings in the main for the respective env variables. Handling quotes, $ and other annoying cases.

### CODE REVIEW

**Comment by mortinso** [comment]
Our minishell doesn't interpret `echo "'$HOME'$PWD"` correctly
image
image

**Comment by mortinso** [comment]
It also doesn't handle `echo $HOME_m` correctly
image
image

**Comment by mortinso** [comment]
`echo $HOME1` is also incorrect
image
image

**Comment by mortinso** [comment]
Notes:
- `_` can be part of an environment value
- numbers can be part of an environment value
- single-quotes should be printed when inside double-quotes

**Comment by mortinso** [comment]
Our minishell SEGV's on `echo "$HOME'a"`
image
image

**Comment by mortinso** [comment]
Need to fix the following situations:

- [x] `$VAR"a"`
- [x] `"'$VAR'"`
- [x] `"$VAR'a"`
- [x] `$"VAR"`
- [x] `$'VAR'`

We should keep testing to make sure its working in all cases. I added a text file with the cases I've been testing for.

**Comment by mortinso** [comment]
We should probably make a separate folder for the replacer

**Comment by ddiniz-m** [comment]
Had errors. Will make a new one

**Comment by mortinso** [comment]
I think all the syntax error functions are fixed now. We should test tomorrow though.

File: src/built-ins/cd.c
Comment (mortinso) [inline src/built-ins/cd.c:1]: cd shouldn't exit

File: src/run/redirections.c
Comment (mortinso) [inline src/run/redirections.c:1]: should actually create an open error function

File: src/signals.c
Comment (mortinso) [inline src/signals.c:1]: should it write exit?

**Review by mortinso — CHANGES_REQUESTED** [review]

**Review by mortinso — APPROVED** [review]

**Pull request merged**